### DISCUSSION TITLE
Backup in-memory SQLITE database

### DISCUSSION THREAD

**Discussion by haoxi911** [mikro-orm/mikro-orm#3937] [Ideas]
I am using Mikro-ORM with in-memory SQLITE database by setting `dbName` to `file::memory:?cache=shared` in `mikro-orm.config.ts` file, and it works! Detailed document [here](https://www.sqlite.org/inmemorydb.html#:~:text=An%20SQLite%20database%20is%20normally,filename%20%22%3Amemory%3A%22.)

I am also looking at a way to dump the in-memory database to a disk file, and I found the document [here](https://www.sqlite.org/backup.html). I think the challenge is how to call these C methods in my Node JS app, any comments / links would be appreciated!

**Comment by B4nan** [comment]
You dont need to use any C code for that, both sqlite drivers should support this

https://stackoverflow.com/questions/40083550/nodejs-persist-a-sqlite-in-memory-database

**Comment by piotr-cz** [comment]
@haoxi911 
How did you manage to make it work?

@mikro-orm/sqlite depends on sqlite3 which [doesn't interpret `file::memory:` correctly](https://github.com/TryGhost/node-sqlite3/issues/1675)

**Reply by haoxi911** [reply]
@piotr-cz Hi, I actually didn't make use of in-memory database in my project right now, so I didn't move further on this topic.
